src/test/java/apitests/DriverTest.java

```diff
@@ -0,0 +1,195 @@
+import com.jdbc.demo.DriverDAO;
+import com.jdbc.demo.domain.psql.Address;
+import com.jdbc.demo.domain.psql.Driver;
+import utils.TestModelsFactory;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = Application.class)
+@WebIntegrationTest
+public class DriverTest {
+
+	private MockMvc mockMvc;
+
+	@Autowired
+	private DriverDAO driverManager;
+
+	@Autowired
+	private AddressDAO addressManager;
+
+	private ArrayList<Driver> driverList = new ArrayList<Driver>();
+	private ArrayList<Address> addressList = new ArrayList<Address>();
+
+	@Autowired
+	private WebApplicationContext webApplicationContext;
+
+	List<Driver> drivers;
+
+	Driver driver1;
+	Driver driver2;
+	Driver driver3;
+
+	private MediaType contentType = new MediaType(MediaType.APPLICATION_JSON.getType(),
+			MediaType.APPLICATION_JSON.getSubtype(), Charset.forName("utf8"));
+
+	@Before
+	public void setup() throws Exception {
+
+		mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build();
+		this.addressList.add(addressManager.add(TestModelsFactory.createTestAddress1()));
+		driver1 = driverManager.add(TestModelsFactory.createTestDriver1(addressList.get(0)));
+		driverList.add(driver1);
+		driver2 = driverManager.add(TestModelsFactory.createTestDriver2(addressList.get(0)));
+		driverList.add(driver2);
+		drivers = driverManager.getAll();
+
+	}
+
+	@After
+	public void cleanup() throws Exception {
+
+		for (Driver driverList : driverList) {
+			driverManager.delete(driverList);
+		}
+	}
+
+	@Test
+	public void getDriver() throws Exception {
+		long LongID = driverList.get(1).getId();
+		int IntID = (int) LongID;
+		
+		mockMvc.perform(get("/api/drivers/" + driverList.get(1).getId()))
+				.andExpect(status().isOk())
+				.andExpect(content().contentType(contentType))
+				.andExpect(jsonPath("$.id", is(IntID)))
+				.andExpect(jsonPath("$.firstName", is(driverList.get(1).getFirstName())))
+				.andExpect(jsonPath("$.lastName", is(driverList.get(1).getLastName())))
+				.andExpect(jsonPath("$.salary", is(driverList.get(1).getSalary().doubleValue())))
+				.andExpect(jsonPath("$.pesel", is(driverList.get(1).getPESEL())))
+				.andExpect(jsonPath("$.available", is(driverList.get(1).isAvailable())))
+				.andExpect(jsonPath("$.deleted", is(driverList.get(1).isDeleted())));
```

Comment:
Instead of calling `driverList.get(1)`, declare local Driver variable and assign value of `driverList.get(1)` to it ( e.g. `Driver testDriver = driverList.get(1);`). I think this would improve readability.